clients/launcher/lib/widgets/welcome_modal.dart

```diff
@@ -110,191 +188,631 @@ class _WelcomeModalContentState extends State<_WelcomeModalContent> {
+                    ),
+                  ),
+                ),
+              ),
             ),
-          ],
+          ),
           const SizedBox(height: 16),
-          Row(
-            mainAxisAlignment: MainAxisAlignment.spaceBetween,
-            children: [
-              SailButton.secondary(
-                _showAdvanced ? 'Simple Mode' : 'Advanced Mode',
-                onPressed: () {
-                  setState(() {
-                    _showAdvanced = !_showAdvanced;
-                  });
-                },
-                size: ButtonSize.regular,
-              ),
-              Row(
+        ],
+      ),
+    );
+  }
+
+  Widget _buildAdvancedScreen() {
+    return Padding(
+      padding: const EdgeInsets.all(4),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Flexible(
+            child: SingleChildScrollView(
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                mainAxisSize: MainAxisSize.min,
                 children: [
-                  SailButton.secondary(
-                    'Generate Random',
-                    onPressed: _handleGenerateRandom,
-                    size: ButtonSize.regular,
+                  // Input field for entropy/hex
+                  Row(
+                    children: [
+                      Expanded(
+                        child: SizedBox(
+                          height: 24,
+                          child: TextField(
+                            controller: _entropyController,
+                            style: const TextStyle(fontSize: 10, color: Colors.white),
+                            decoration: const InputDecoration(
+                              isDense: true,
+                              contentPadding: EdgeInsets.symmetric(horizontal: 4, vertical: 4),
+                              hintText: 'Enter any text or hex',
+                              hintStyle: TextStyle(fontSize: 10, color: Colors.white70),
+                              border: OutlineInputBorder(
+                                borderRadius: BorderRadius.all(Radius.circular(2)),
+                              ),
+                              enabledBorder: OutlineInputBorder(
+                                borderRadius: BorderRadius.all(Radius.circular(2)),
+                                borderSide: BorderSide(color: Colors.white24),
+                              ),
+                              focusedBorder: OutlineInputBorder(
+                                borderRadius: BorderRadius.all(Radius.circular(2)),
+                                borderSide: BorderSide(color: Colors.white),
+                              ),
+                            ),
+                            onChanged: _handleEntropyInput,
+                          ),
+                        ),
+                      ),
```

Comment:
Should user SailTextField here. Reason being that this will only work in dark mode, but unusable in dark mode. If SailTextField needs tweaks and changes, lets perfect that!